### ISSUE TITLE
REST API - Exclude taxes option from earnings

### ISSUE THREAD

**Issue by mintplugins** [easydigitaldownloads/easy-digital-downloads#5444]
A customer has requested that the earnings endpoint of the rest API have an option include or exclude taxes. 

Ticket:
https://secure.helpscout.net/conversation/315603335/52472/?folderId=180499

**Comment by sunnyratilal** [comment]
We can get this in as a simple fix for 2.7.3.

**Comment by cklosowski** [comment]
Ok this is fixed up, with a single caviate, the default with no date supplied, for the total earnings will always include tax, since the function `edd_get_total_earnings()` actually does not use the EDD Payment Stats class which includes tax. but the date ranged requests do now support a query string of the following:

-[] `include_tax=1`
-[] `include_tax=0`
-[] `include_tax=true`
-[] `include_tax=false`
-[] `include_tax=yes`
-[] `include_tax=no`

By default, all returns include tax.

**Comment by mintplugins** [comment]
I'm getting this error from this endpoint:
https://edd2.dev/sl/edd-api/stats/?key=api_key&token=token_key

```

( ! ) Warning: Missing argument 1 for EDD_API::get_default_earnings_stats(), called in /app/public/wp-content/plugins/easy-digital-downloads/includes/api/class-edd-api.php on line 1454 and defined in /app/public/wp-content/plugins/easy-digital-downloads/includes/api/class-edd-api.php on line 2200
```

**Comment by cklosowski** [comment]
Punting until 3.0 is out, so we can _maybe_ support removing taxes from `edd_get_total_earnings()`

**Comment by cklosowski** [comment]
Punting until after 3.0 so we can _maybe_ support removing taxes in `edd_get_total_earnings()`

**Comment by mintplugins** [comment]
My concerns, which led to this being punted, are that when you set `include_tax` to `false`, the `totals` number, when not using a date range, include taxes no matter what, and give no indication that they do.

**Comment by sunnyratilal** [comment]
This will now be possible due to the way @JJJ and I have structured orders. I'll make PR once we've merged #4576.

**Comment by sunnyratilal** [comment]
PR refreshed and merged. `edd_get_total_earnings()` now has an `$include_taxes` parameter. EDD API updated to use the new parameter.
